Sources/KindeSDK/Auth/Auth.swift

```diff
@@ -107,6 +247,82 @@ public final class Auth {
         return nil
     }
     
+    public func getRoles(options: ApiOptions? = nil) async throws -> Roles {
+        if options?.forceApi == true {
+            let response = try await RolesAPI.getRoles()
+            guard response.success, let data = response.data else {
+                throw NSError(domain: "KindeSDK", code: -1, userInfo: [NSLocalizedDescriptionKey: "Failed to fetch roles from API - check network and authentication"])
+            }
+            let orgCode = data.orgCode ?? ""
+            let organization = Organization(code: orgCode)
+            return Roles(organization: organization, roles: response.getRoleKeys())
+        } else {
+            if let rolesClaim = getClaim(forKey: ClaimKey.roles.rawValue),
+               let rolesArray = rolesClaim.value as? [String],
+               let orgCodeClaim = getClaim(forKey: ClaimKey.organisationCode.rawValue),
+               let orgCode = orgCodeClaim.value as? String {
+                
+                let organization = Organization(code: orgCode)
+                let roles = Roles(organization: organization,
+                                 roles: rolesArray)
+                return roles
+            }
+            throw NSError(domain: "KindeSDK", code: -1, userInfo: [NSLocalizedDescriptionKey: "Roles not found in token claims"])
+        }
+    }
+    
+    /// Get all roles for the authenticated user (synchronous version, reads from token claims)
+    /// - Returns: Roles if found, nil otherwise
+    public func getRoles() -> Roles? {
+        if let rolesClaim = getClaim(forKey: ClaimKey.roles.rawValue),
+           let rolesArray = rolesClaim.value as? [String],
+           let orgCodeClaim = getClaim(forKey: ClaimKey.organisationCode.rawValue),
+           let orgCode = orgCodeClaim.value as? String {
+            
+            let organization = Organization(code: orgCode)
+            let roles = Roles(organization: organization,
+                             roles: rolesArray)
+            return roles
+        }
+        return nil
+    }
+    
+    public func getRole(name: String, options: ApiOptions? = nil) async throws -> Role {
+        if options?.forceApi == true {
+            let roles = try await getRoles(options: options)
+            return Role(organization: roles.organization, isGranted: roles.roles.contains(name))
+        } else {
+            if let rolesClaim = getClaim(forKey: ClaimKey.roles.rawValue),
+               let rolesArray = rolesClaim.value as? [String],
+               let orgCodeClaim = getClaim(forKey: ClaimKey.organisationCode.rawValue),
+               let orgCode = orgCodeClaim.value as? String {
+                
+                let organization = Organization(code: orgCode)
+                let role = Role(organization: organization,
+                               isGranted: rolesArray.contains(name))
+                return role
+            }
```

Comment:
Since the same logic is in the `getRole(name: String)` function, we can use that directly here.

```suggestion
            if let role = getRole(name: name) {
                return role
            }
```